expression/builtin_time_vec.go

```diff
@@ -724,11 +724,51 @@ func (b *builtinLastDaySig) vecEvalTime(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinAddDateStringDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinAddDateStringDecimalSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	unit, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(unit)
+	if err = b.args[2].VecEvalString(b.ctx, input, unit); err != nil {
+		return err
+	}
+	result.ResizeTime(n, false)
+	result.MergeNulls(unit)
+	results := result.Times()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		unitI := unit.GetString(i)
+		row := input.GetRow(i)
+		date, isNull, err := b.getDateFromString(b.ctx, b.args, row, unitI)
+		if err != nil {
+			return err
+		}
+		if isNull {
+			result.SetNull(i, true)
+			continue
+		}
+		interval, isNull, err := b.getIntervalFromDecimal(b.ctx, b.args, row, unitI)
+		if err != nil {
+			return err
+		}
+		if isNull {
+			result.SetNull(i, true)
+			continue
+		}
+		resultI, _, err := b.add(b.ctx, date, interval, unitI)
```

Comment:
Here we should get the ```isNull```, and check the ```isNull```.

Reply:
done